vue/components/ui/molecules/code-editor/code-editor.vue

```diff
@@ -0,0 +1,209 @@
+                    v-for="line in linesInfo"
+                    v-bind:key="line.number"
+                >
+                    <tooltip
+                        v-bind:text="line.error.error"
+                        v-bind:variant="'red'"
+                        v-bind:orientation="'top'"
+                        v-bind:alignment="'left'"
+                        v-bind:text-alignment="'left'"
+                        v-bind:width="250"
+                        v-bind:visible="true"
+                        v-bind:border-radius="'6px'"
+                        v-bind:delay="100"
+                        v-if="line.error"
+                    >
+                        <span class="number">{{ line.number }}</span>
+                    </tooltip>
+                    <span class="number" v-else>{{ line.number }}</span>
+                    <span class="text">{{ line.line }}</span>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<style lang="scss" scoped>
+@import "css/variables.scss";
+
+.code-editor {
+    position: relative;
+}
+
+.code-editor > textarea,
+.code-editor > .overlay {
+    border: 1px solid $light-white;
+    border-radius: 6px 6px 6px 6px;
+    box-sizing: border-box;
+    outline: none;
+    padding: 8px 12px 8px 0px;
+}
+
+.code-editor > .textarea {
+    padding-left: 44px;
+}
+
+.code-editor > .overlay {
+    font-size: 0px;
+    height: 100%;
+    left: 0;
+    padding-top: 9px;
+    pointer-events: none;
+    position: absolute;
+    top: 0;
+    width: 100%;
+}
+
+.code-editor > textarea,
+.code-editor > .overlay > .lines > .line .number,
+.code-editor > .overlay > .lines > .line > .text {
+    font-family: "consolas", monospace;
+    font-size: 13px;
+    font-weight: 500;
+    letter-spacing: 0px;
+    line-height: 18px;
+}
+
+.code-editor > .overlay > .lines > .line .number,
+.code-editor > .overlay > .lines > .line > .text {
+    display: inline-block;
+}
+
+.code-editor > .overlay > .lines > .line .number {
+    border-right: transparent solid 2px;
+    box-sizing: border-box;
+    color: #979797;
+    margin: 0px 4px 0px 0px;
+    padding: 0px 2px 0px 0px;
+    pointer-events: all;
+    text-align: right;
+    width: 40px;
+}
+
+.code-editor > .overlay > .lines > .line.error .number {
+    border-right: #ae2929 solid 2px;
+}
+
+.code-editor > .overlay > .lines > .line > .text {
+    color: transparent;
+    white-space: pre;
+}
+</style>
+
+<script>
+import { normalize } from "ripe-commons";
+
+export const CodeEditor = {
+    name: "code-editor",
+    props: {
+        value: {
+            type: String,
+            default: null
+        },
+        disabled: {
+            type: Boolean,
+            default: false
+        },
+        type: {
+            type: String,
+            default: "json"
+        },
+        validate: {
+            type: Function,
+            default: null
+        },
+        debounceTime: {

```

Comment:
Leave a description of what this is.
Also, isn't the default value a bit too low?

Reply:
It's literally the a debounce time. The value is good, I tried different ones and this felt right. It's high enough to do what is supposed to do and short enough to be responsive.